ui/ui-components/components/layouts/main.tsx

```diff
@@ -7,9 +7,21 @@ import ErrorAlert from "../alerts/error";
 import Navigation from "../navigation";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import HydrationError from "../alerts/hydrationError";
+import { SuccessHandler } from "../../utils/successHandler";
+import { ErrorHandler } from "../../utils/errorHandler";
 
 export default function Main(props) {
-  const { children, successHandler, errorHandler, hydrationError } = props;
+  const {
+    children,
+    successHandler,
+    errorHandler,
+    hydrationError,
+  }: {
+    children: any;
+    successHandler: SuccessHandler;
+    errorHandler: ErrorHandler;
+    hydrationError: string;
```

Comment:
We may want to refactor some around `hydrationError`s.  Sometimes we use it in reference to the string message itself, and sometimes we use it in reference to the `Error`.  I considered that outside of this PR.

Reply:
Good find, and I agree! Can you write up a story in Tracker please about this?